sleap/nn/inference.py

```diff
@@ -5461,43 +5490,169 @@ def main(args: Optional[list] = None):
     print()
 
     # Setup data loader.
-    provider, data_path = _make_provider_from_cli(args)
+    provider_list, data_path_list = _make_provider_from_cli(args)
 
     # Setup tracker.
     tracker = _make_tracker_from_cli(args)
 
+    output_path = args.output
+
     if args.models is not None and "movenet" in args.models[0]:
         args.models = args.models[0]
 
-    # Either run inference (and tracking) or just run tracking
+    # Either run inference (and tracking) or just run tracking (if using an existing prediction where inference has already been run)
     if args.models is not None:
-        # Setup models.
-        predictor = _make_predictor_from_cli(args)
-        predictor.tracker = tracker
 
-        # Run inference!
-        labels_pr = predictor.predict(provider)
+        # Run inference on all files inputed
+        for data_path, provider in zip(data_path_list, provider_list):
+            # Setup models.
+            data_path_obj = Path(data_path)
+            predictor = _make_predictor_from_cli(args)
+            predictor.tracker = tracker
+
+            # Run inference!
+            labels_pr = predictor.predict(provider)
+
+            # if output path was not provided, create an output path
+            if output_path is not None:
+                print("HERE")
+                output_path_obj = Path(output_path)
+                print(output_path_obj.exists())
+                print(output_path_obj.is_dir())
+                if (
+                    output_path_obj.exists()
+                    and output_path_obj.is_file()
+                    and len(data_path_list) > 1
+                ):
+                    raise ValueError(
+                        "output_path argument must be a directory if multiple video inputs are given"
+                    )
 
-        if output_path is None:
-            output_path = data_path + ".predictions.slp"
+            if output_path is None:
+                output_path = f"{data_path.as_posix()}.predictions.slp"
+                output_path_obj = Path(output_path)
 
-        labels_pr.provenance["model_paths"] = predictor.model_paths
-        labels_pr.provenance["predictor"] = type(predictor).__name__
+            # if output_path was provided and multiple inputs were provided, create a directory to store outputs
+            elif len(data_path_list) > 1:
+                output_path = (
+                    output_path_obj.as_posix()
+                    + "/"
+                    + (data_path_obj.stem + ".predictions.slp")
+                )
```

Comment:
Rather than using string operations, with pathlib you can use the `/` binary operator to concatenate path segments:

```suggestion
                output_path = output_path_obj / data_path_obj.with_suffix(".predictions.slp")
```